contrib/babelfishpg_tsql/src/pltsql_bulkcopy.c

```diff
@@ -979,6 +1013,48 @@ EndBulkCopy(BulkCopyState cstate, bool aborted)
 {
 	if (cstate)
 	{
+		/* 
+		 * for identity column reseed the identity sequence value, 
+		 * if cur_identity_value is more suitable as compared to previous seed value.
+		 */
+		if (cstate->seqid != InvalidOid && !aborted)
+		{
+			/* calculate the previous seed value */
+			int64	init_identity_value = 1;
+			ListCell	*seq_lc;
+			List	*seq_options = sequence_options(cstate->seqid);
+
+			foreach(seq_lc, seq_options)
+			{
+				DefElem    *defel = (DefElem *) lfirst(seq_lc);
+
+				if (strcmp(defel->defname, "start") == 0)
+				{
+					init_identity_value = defGetInt64(defel);
+					break;
+				}
+			}
+
+			PG_TRY();
+			{
+				init_identity_value = DirectFunctionCall1(pg_sequence_last_value,
```

Comment:
Our implementation in get_identity_current is the other way around - if we get an error returning from pg_sequence_last_value then only we set ``init_identity_value = defGetInt64(defel);`` 

Which seems to be better, we only check for sequence_options when pg_sequence_last_value throws an error.

The comment also describes it:
```
 * Given a table name with an identity column, fetch the last identity sequence
 * value stored in the pg_sequences catalog. If not set, return the seed value
 * instead. Return NULL on error.
 ```

Reply:
Makes sense, reversed the order